### DISCUSSION TITLE
Question about "per-module" mode in vite package

### DISCUSSION THREAD

**Discussion by kingcw** [unocss/unocss#1335] [Q&A] [answered]
In doc, it said "per-module" mode is to generate css for each module, I was expecting one css file per module.

But after generation, there's still only one style.css file, and a "@unocss" folder with a bunch of ${hash}.mjs files with just following 4 lines.

`2b2e9c74.mjs`
```js
const e = "";
export {
  e as default
};
```

Each unocss generated mjs file is imported per actual code mjs file

`Button.mjs`
```js
import "../@unocss/8077097a.mjs";
...
```

I have a hard time undestanding what it does, I thought "per-module" is used for css tree shaking, like rollup output options "preserveModules" that output one mjs file per module

Can someone clarify the concept of "per-module"?  What it does and what it's used for?

Thank you.

**Comment by antfu** [comment, marked as answer]
UnoCSS generates css for each mode, while Vite's build process will combine multiple CSS into a single one in the same chunk. A new CSS file will be generated only if there are in different chunks just like your js code.

**Reply by kingcw** [reply]
Is "generating css file per module (per js/ts file)" doable in unocss alone without modifying vite?  If so, I'd like to contribute to make this work.

Thank you

**Reply by antfu** [reply]
TBH, I'd not recommend using it. The order of CSS in multiple styles can not be guaranteed (at least before the native CSS layer is landed widely), which will cause prioritize issues in navigating MPA.
